pkg/azuredisk/azuredisk_v2.go

```diff
@@ -114,9 +141,27 @@ func (d *DriverV2) Run(endpoint, kubeconfig string, testBool bool) {
 		csi.NodeServiceCapability_RPC_GET_VOLUME_STATS,
 	})
 
+	// Start the controllers if this is a controller plug-in
+	if *isControllerPlugin {
+		go controller.StartControllersAndWaitForExit(d.kubeConfig, &d.azVolumeClient, "default", d.objectNamespace)
+	}
+
+	// Register the AzDriverNode
+	if *isNodePlugin {
+		RegisterAzDriverNodeOrDie(d.NodeID, d.kubeClient, d.azVolumeClient, "default", d.objectNamespace)
+	}
+
+	// Start the CSI endpoint/server
 	s := csicommon.NewNonBlockingGRPCServer()
-	// Driver d act as IdentityServer, ControllerServer and NodeServer
+	// Driver d acts as IdentityServer, ControllerServer and NodeServer
 	s.Start(endpoint, d, d, d, testBool)
+
+	// Start sending hearbeat and mark node as ready
+	if *isNodePlugin {
+		go RunAzDriverNodeHeartbeatLoop(d.NodeID, d.kubeClient, d.azVolumeClient, "default", d.objectNamespace)
```

Comment:
Can this be combined in the previous `if` statement registering the node?

Reply:
I added a response to the other comment, let me know if it makes sense, otherwise I'll change it.